packages/lit-html/src/lib/lit-html.ts

```diff
@@ -560,6 +580,119 @@ export class NodePart {
   }
 }
 
+export class AttributePart {
+  readonly __element: HTMLElement;
+  readonly name: string;
+
+  /**
+   * If this attribute part represents an interpolation, this contains the
+   * static strings of the interpolation. For single-value, complete bindings,
+   * this is undefined.
+   */
+  readonly __strings?: ReadonlyArray<string>;
+  __value: unknown | Array<unknown> = nothing;
+
+  constructor(
+    element: HTMLElement,
+    name: string,
+    strings: ReadonlyArray<string>,
+    _options?: RenderOptions
+  ) {
+    this.__element = element;
+    this.name = name;
+    if (strings.length > 2 || strings[0] !== '' || strings[1] !== '') {
+      this.__value = new Array(strings.length - 1).fill(nothing);
+      this.__strings = strings;
+    }
```

Comment:
```suggestion
    } else {
      this.__value = nothing;
    }
```

Seems like we should initialize single-value bindings' value to `nothing` so that an initial render of `nothing` dirty checks to no change?